File: 1435-xor-queries-of-a-subarray/xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = arr.size();
        
        vector<int> prefix;
        prefix.push_back(arr[0]);
        for (int i=1; i<n; i++) {
            prefix.push_back(arr[i]);
            prefix[i] = prefix[i] ^ prefix[i-1];
        }
        
        vector<int> ans;
        for (int i=0; i<queries.size(); i++) {
            int min_ = queries[i][0];
            int max_ = queries[i][1];
            int curr_ans = min_ == 0 ? prefix[max_] : prefix[max_] ^ prefix[min_ - 1];
            ans.push_back(curr_ans);
        }
        
        return ans;
    }
};

File: 1435-xor-queries-of-a-subarray/xor-queries-of-a-subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        prefix_xor = [0] * (len(arr) + 1)
        for i in range(1, len(arr)+1):
            prefix_xor[i] = arr[i-1] ^ prefix_xor[i-1]
        
        ans = [prefix_xor[r+1] ^ prefix_xor[l] for l,r in queries]
        return ans